### ISSUE TITLE
TypeError: Could not convert argument 1 to a number

### ISSUE THREAD

**Issue by RobertBaron** [feross/buffer#123]
Getting
```js
TypeError: Could not convert argument 1 to a number
```

With function fromArrayBuffer:256
```js
array = new Uint8Array(array, byteOffset)
```

On 
 - Android 4.4.4, Android 4.2.2
 - UC Browser 10.10.5, 10.10.0, 10.9.8

**Comment by feross** [comment]
@dcousens Is this a bug you were able to confirm? I usually reserve the `bug` label for bugs that I can actually reproduce.

@RobertBaron Do you have a code example that can consistently reproduce the error?

**Comment by dcousens** [comment]
@feross no,  I don't have an Android device or environment at those versions.  Removed tag.

**Comment by feross** [comment]
@RobertBaron UC Browser is not a supported browser. Sauce Labs doesn't even offer that for testing.

I was able to reproduce the issue you described on Android 4.2 and 4.3 with the following code:

```js
var ab = new ArrayBuffer(4)
Buffer.from(ab)
```

It happens because apparently the `Uint8Array` constructor on those versions of Android Browser can't handle the second argument being `undefined`. Instead the second argument (which is optional) must be omitted entirely instead of passed in as `undefined`.

Like this:

```js
  if (byteOffset === undefined && length === undefined) {
    array = new Uint8Array(array)
  } else if (length === undefined) {
    array = new Uint8Array(array, byteOffset)
  } else {
    array = new Uint8Array(array, byteOffset, length)
  }
```

Fixed 👍 Will be released as 4.7.1.
